datastore/datastore.go

```diff
@@ -203,7 +209,10 @@ func newClient(scope string, kv string, addr string, config *store.Config, cache
 		config = &store.Config{}
 	}
 
-	addrs := strings.Split(addr, ",")
+	// Add the custom prefix to the root chain
+	if len(parts) == 2 {
+		rootChain = append([]string{parts[1]}, defaultRootChain...)
+	}
```

Comment:
Also, does the discovery support just 1 prefix for a given store (or) different prefixes for different stores (since it supports a comma separated list of store urls).